org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/template/java/CodeSnippetTemplate.java

```diff
@@ -110,7 +110,7 @@ class TemplatePreferences {
 	public static final String TRYCATCH_CONTENT = "try {\n" + "\t$$TM_SELECTED_TEXT$${1}\n" + "} catch ($${2:Exception} $${3:e}) {\n" + "\t$${0}// TODO: handle exception\n" + "}";
 	public static final String TRYRESOURCES_CONTENT = "try ($${1}) {\n" + "\t$$TM_SELECTED_TEXT$${2}\n" + "} catch ($${3:Exception} $${4:e}) {\n" + "\t$${0}// TODO: handle exception\n" + "}";
 	public static final String MAIN_CONTENT = "public static void main(String[] args) {\n" + "\t$${0}\n" + "}";
-	public static final String CTOR_CONTENT = "$${1|public,protected,private|} $${2:$$TM_FILENAME_BASE}($${3}) {\n" + "\t$${4:super();}$${0}\n" + "}";
+	public static final String CTOR_CONTENT = "$${1|public,protected,private|} REPLACE_CLASS_NAME($${2}) {\n" + "\t$${3:super();}$${0}\n" + "}";
```

Comment:
Is it possible to use the variables already defined in places like https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/f4d1a4a3d31665f23bb5422fe1b99ce823d76f25/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/CodeTemplatePreferences.java#L196 with `${type_name}` . You basically would just need to pass the template first to some evaluate function to properly resolve those, and then resolve the LSP-based ones.

Could that work. Otherwise, this is sort of just re-implementing functionality we already have.